tests/test_arithmetics.rs

```diff
@@ -423,3 +423,226 @@ fn rich_comparisons_python_3_type_error() {
+            lhs
+        }
+
+        fn __iadd__(&'p mut self, _other: PyRef<'p, Self>) {}
+        fn __isub__(&'p mut self, _other: PyRef<'p, Self>) {}
+        fn __imul__(&'p mut self, _other: PyRef<'p, Self>) {}
+        fn __imatmul__(&'p mut self, _other: PyRef<'p, Self>) {}
+        fn __itruediv__(&'p mut self, _other: PyRef<'p, Self>) {}
+        fn __ifloordiv__(&'p mut self, _other: PyRef<'p, Self>) {}
+        fn __imod__(&'p mut self, _other: PyRef<'p, Self>) {}
+        fn __ipow__(&'p mut self, _other: PyRef<'p, Self>) {}
+        fn __ilshift__(&'p mut self, _other: PyRef<'p, Self>) {}
+        fn __irshift__(&'p mut self, _other: PyRef<'p, Self>) {}
+    }
+
+    fn _test_bool_operator(operator: &str) {
+        let gil = Python::acquire_gil();
+        let py = gil.python();
+        let c2 = PyCell::new(py, RichComparisonToSelf {}).unwrap();
+        py_run!(
+            py,
+            c2,
+            &format!(
+                "\
+class Other:
+    def __eq__(self, other):
+        return True
+    __ne__ = __lt__ = __le__ = __gt__ = __ge__ = __eq__
+
+assert (c2 {} Other()) is True",
+                operator
+            )
+        );
+    }
+
+    fn _test_logical_operator(operator: &str) {
+        _test_bool_operator(operator);
+
+        let gil = Python::acquire_gil();
+        let py = gil.python();
+        let c2 = PyCell::new(py, RichComparisonToSelf {}).unwrap();
+        py_expect_exception!(
+            py,
+            c2,
+            &format!("class Other: pass\nc2 {} Other()", operator),
+            PyTypeError
+        )
+    }
+
+    fn _test_binary_num_operator(operator: &str) {
+        let gil = Python::acquire_gil();
+        let py = gil.python();
+        let c2 = PyCell::new(py, RichComparisonToSelf {}).unwrap();
+        py_run!(
+            py,
+            c2,
+            &format!(
+                "\
+class Other:
+    def __radd__(self, other):
+        return other
+    __rand__ = __ror__ = __rxor__ = __radd__
+    __rsub__ = __rmul__ = __rtruediv__ = __rfloordiv__ = __rpow__ = __radd__
+    __rmatmul__ = __rlshift__ = __rrshift__ = __rmod__ = __rdivmod__ = __radd__
+
+assert (c2 {} Other()) is c2",
+                operator
+            )
+        );
+
+        py_expect_exception!(
+            py,
+            c2,
+            &format!("class Other: pass\nc2 {} Other()", operator),
+            PyTypeError
+        )
+    }
+
+    fn _test_inplace_num_operator(operator: &str) {
+        let gil = Python::acquire_gil();
+        let py = gil.python();
+        let c2 = PyCell::new(py, RichComparisonToSelf {}).unwrap();
+        py_run!(
+            py,
+            c2,
+            &format!(
+                "\
+class Other:
+    def __radd__(self, other):
+        return other
+    __rand__ = __ror__ = __rxor__ = __radd__
+    __rsub__ = __rmul__ = __rtruediv__ = __rfloordiv__ = __rpow__ = __radd__
+    __rmatmul__ = __rlshift__ = __rrshift__ = __rmod__ = __rdivmod__ = __radd__
```

Comment:
I don't know whether it's worth it, but if you wanted to make only one definition of `Other` you might be able to do a helper function like:

```
def get_other_type(py: Python) -> &PyAny {
    let locals = PyDict::new(py);
    py.run("""
class Other:
    # put all operators here
""", None, Some(locals)).unwrap();
    locals.get_item("Other").unwrap()
}

Reply:
Nice.  Done.